### ISSUE TITLE
DOC: underspecified (byte-array)  and <byte-array> 

### ISSUE THREAD

**Issue by nomennescio** [factor/factor#2617]
Documentation of words [`(byte-array)`](https://docs.factorcode.org/content/word-%28byte-array%29%2Cbyte-arrays.html)   and [`<byte-array>`](https://docs.factorcode.org/content/word-__lt__byte-array__gt__%2Cbyte-arrays.html) in `byte-arrays` seem underspecified:

The first has
```
Word description
Creates a new byte array with unspecified contents of length n bytes.

Definition

PRIMITIVE: (byte-array) ( n -- byte-array ) flushable
```

The second
```
Word description
Creates a new byte array holding n bytes.

Definition
PRIMITIVE: <byte-array> ( n -- byte-array ) flushable
```

If the former has unspecified contents, the latter has specified contents? If so, what is it?

**Comment by timor** [comment]
`<byte-array>` sets all fields to `0`.

**Comment by mrjbq7** [comment]
I agree, we should differentiate between allocated and unspecified and allocated and initialized to zero in the docs. I can make that update soon.
